### PR TITLE
Debugging is built and tested with llvm 3.3.

### PR DESCRIPTION
**Pull request by hzhang86**
Also built with llvm 3.6, but test results verification error.
Added test program and python test script in test/llvmDebug

### CODE REVIEW

**Comment by hzhang86** [comment]
This one should be ready to merge.

File: make/platform/Makefile.linux64 [hunk 1]
```diff
@@ -17,3 +17,5 @@
 
 # Makefile.linux64
 # Special AR and RANLIB definitions for XMT used to be here.
+
+CXXFLAGS += -frtti
```
Comment (mppf) [inline make/platform/Makefile.linux64:21]: Why is this change here? This should not be needed. You should be getting -frtti from third-party/llvm/Makefile.include-llvm - and if you're not, we need to fix that file instead.

File: third-party/llvm/Makefile.include-llvm [hunk 2]
```diff
@@ -17,5 +17,5 @@ LLVM_CLANG_LIBS=-lclangFrontend -lclangSerialization -lclangDriver -lclangCodeGe
 LLVM_CXXFLAGS=$(shell $(LLVM_INSTALL_DIR)/bin/llvm-config --cxxflags | sed 's/ -O[0-4s] / /') -DHAVE_LLVM
 LLVM_CFLAGS=$(shell $(LLVM_INSTALL_DIR)/bin/llvm-config --cflags | sed 's/ -O[0-4s] / /') -DHAVE_LLVM
 LLVM_INCLUDES=-I$(LLVM_INCLUDE_DIR) -I$(LLVM_CLANG_CODEGEN_INCLUDE_DIR)
-LLVM_LIBS=-L$(LLVM_LIB_DIR) $(LLVM_CLANG_LIBS) $(LLVM_LLVM_LIBS)
+LLVM_LIBS=-L$(LLVM_LIB_DIR) $(LLVM_CLANG_LIBS) $(LLVM_LLVM_LIBS) -lLLVMSupport -ldl
```
Comment (mppf) [inline third-party/llvm/Makefile.include-llvm:20]: These changes are in the right file - but I suspect that -lLLVMSupport -ldl is only needed for 3.3. Why did you add these?

Comment (mppf) [inline third-party/llvm/Makefile.include-llvm:20]: In particular - these libraries should be gotten from this line:

 # Use this version for LLVM 3.4 and 3.3
 #LLVM_LLVM_LIBS=$(shell $(LLVM_INSTALL_DIR)/bin/llvm-config --ldflags --libs bitreader bitwriter ipo instrumentation option objcarcopts all-targets)

which you're supposed to uncomment for use with LLVM 3.3 (and comment out the 3.6 version)

File: util/config/gather-clang-sysroot-arguments [hunk 3]
```diff
@@ -7,7 +7,8 @@ SYS_BASIC="$CHPL_HOME/runtime/include/sys_basic.h"
 
 SYSTEM_CLANG_COMMANDS=`clang -### $SYS_BASIC 2>&1`
 
-if [ $? -eq 0 ]; then
+#if [ $? -eq 0 ]; then
+if []; then
```
Comment (mppf) [inline util/config/gather-clang-sysroot-arguments:11]: Why is this here? Did this not work with LLVM 3.3?

Comment (mppf) [inline util/config/gather-clang-sysroot-arguments]: Better to tell me about the problem so I can fix it correctly than to

File: test/llvmDebug/llvmDebug_test.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+# Verify the formal arguments
+if 'My_level' in output:
+    if 'My_x' in output:
+        print 'checking formal arguments --PASS'
+else:
+    print 'checking formal arguments --FAIL'
+
+print 'DONE with debug checking'
+
```
Comment (mppf) [inline test/llvmDebug/llvmDebug_test.py:71]: This test script looks pretty good, although I havn't run it, and it's specific to linux64-gnu (because of the llvm_tool_path). But that's OK for now for this test.

**Comment by mppf** [comment]
I'm going to merge this and then take out some of the changes so we can get to the same version to work towards 3.6/3.7 support.

File: make/platform/Makefile.linux64 (hunk 1 above)
Comment (hzhang86) [inline make/platform/Makefile.linux64:21]: that's just a shortcut I took when I tried to build 3.6, not happened to 3.3

Comment (mppf) [inline make/platform/Makefile.linux64:21]: I build 3.6 all the time (with the right part of Makefile.llvm-include commented out) and don't need this change

File: third-party/llvm/Makefile.include-llvm (hunk 2 above)
Comment (hzhang86) [inline third-party/llvm/Makefile.include-llvm:20]: I was building 3.6 so I commented out this line

Yes, they are needed for 3.3, I thought 3.6 needs them too, so I didn't remove them when I built 3.6

File: util/config/gather-clang-sysroot-arguments (hunk 3 above)
Comment (hzhang86) [inline util/config/gather-clang-sysroot-arguments:11]: i ran into an old clang reference problem when I was trying to build chapel with llvm 3.3, and I found the problem:
/export/home/hzhang86/chapel/chapel/third-party/llvm/install/linux64-gnu/bin/clang -resource-dir /usr/bin/../lib/clang/3.0 -o conftest -O3   conftest.c  >&5

therefore I manually edit this, I think you should be fine if no old clang installed

File: third-party/llvm/Makefile.include-llvm (hunk 2 above)
Comment (hzhang86) [inline third-party/llvm/Makefile.include-llvm:20]: I was building 3.6, so I commented it line out.

Yes, they are needed for 3.3, so when i built 3.6, I thought they are needed as well, I left them there.

File: make/platform/Makefile.linux64 (hunk 1 above)
Comment (hzhang86) [inline make/platform/Makefile.linux64:21]: not sure about that, if I don't add this, it'll fail with some dynamic functions, like isa<>()...

Comment (mppf) [inline make/platform/Makefile.linux64:21]: LLVM builds with -fno-rtti and you should too. Are you using isa<> ? You should be using the LLVM equivalent.

Comment (mppf) [inline make/platform/Makefile.linux64:21]: There are problems mixing RTTI and no-RTTI code - and LLVM wil build with no-RTTI.
